zebra-network/src/peer_set/initialize.rs

```diff
@@ -211,7 +225,27 @@ where
     S: Service<(TcpStream, SocketAddr), Response = peer::Client, Error = BoxError> + Clone,
     S::Future: Send + 'static,
 {
-    let listener = TcpListener::bind(addr).await?;
+    println!("attempting to open listener");
+    let listener_fut = TcpListener::bind(addr);
```

Comment:
The listener bind has a long timeout on some platforms.
Should we wrap bind in spawn_blocking, so the listener bind gets its own thread, and doesn't block the timeout task?

Reply:
Let's also log an info-level message outside the spawn_blocking, to help us diagnose bugs